src/signal_routing.rs

```diff
@@ -52,6 +52,59 @@ pub fn route_action(
     route_signal(action, signal, action.name().as_str(), target)
 }
 
+fn panic_if_signal_cannot_be_queued(signal_name: &str, parameters: &[glib::Value]) {
+    for (i, param) in parameters.iter().enumerate() {
+        let param_type = param.type_();
+        if param_type.name().ends_with("Context") {
```

Comment:
How many of these are there? I'd prefer an explicit list to be honest. Also, some public API to hack into that list would be cool.

Reply:
I wish I had such a list, but I can't even find a proper list of all possible signals. This is a heuristic that seems to hold pretty well - from [a quick search (that requires some manual post-filtering)](https://gtk-rs.org/gtk3-rs/stable/latest/docs/gtk/?search=Context) I can find some more signals, that I did not know of, that have `Context` parameters and I think verify my heuristic:

* [The `draw-page` signal](https://developer.gnome.org/gtk3/stable/gtk3-High-level-Printing-API.html#GtkPrintOperation-draw-page) has a `GtkPrintContext` parameter, and queuing this signal means you'll draw after the page was printed.
* [The `render` signal](https://developer.gnome.org/gtk3/stable/GtkGLArea.html#GtkGLArea-render) has a `GdkGLContext` parameter, and I think queuing this signal will have similar results to queuing `draw`.
* I could not find the signal that accepts [`DragContext`](https://gtk-rs.org/gtk3-rs/stable/latest/docs/gdk/struct.DragContext.html), but it has methods like `drop_finish` and `drop_abort` that suggest you can't queue it.